### PR TITLE
Remove cppcheck suppressions and fix errors/warnings

### PR DESCRIPTION
**Pull request by tari01**

### CODE REVIEW

File: src/menu.cpp [hunk 1]
```diff
@@ -107,10 +107,6 @@ Menu::get_display_appointments(const std::vector<Appointment>& appointments_in,
             if (a_full_day_today != b_full_day_today)
                 return a_full_day_today;
 
-            const bool a_after_today = (a.begin > end_of_day) || (a.end > end_of_day);
-            const bool b_after_today = (a.begin > end_of_day) || (a.end > end_of_day);
-            if (a_after_today != b_after_today)
-                return a_after_today;
```
Comment (sunweaver) [inline src/menu.cpp:113]: These changes feel like you lost some code in this commit. Please have this code removal as a separate commit / PR and explain better why you think removing this is good.

File: tests/print-to.h [hunk 2]
```diff
@@ -27,63 +27,6 @@
...
-{
-    *os << '{';
-    *os << "{text:" << alarm.text << '}';
-    PrintTo(alarm.time, os);
-    *os << '}';
-}
-
-void
-PrintTo(const Appointment& appointment, std::ostream* os)
-{
```
Comment (sunweaver) [inline tests/print-to.h:55]: Simply sneaking in a test-removal into this commit is not appropriate. Separate commit, please with good explanation on why you do this.

File: tests/test-menu-appointments.cpp [hunk 3]
```diff
@@ -117,70 +116,6 @@ TEST_F(MenuAppointmentFixture, DisplayEvents)
...
-    );
-
-    const auto pub_with_pawel = create_appointment(
-        Appointment::ALARM,
-        "uid-pawel",
-        "Meet Pawel at the Pub",
-        DateTime::Local(2017,2,4,19,0,0),
-        DateTime::Local(2017,2,4,19,0,0)
-    );
-
```
Comment (sunweaver) [inline tests/test-menu-appointments.cpp:183]: also a massive code removal without proper explanations... Please amend.

**Review by sunweaver — CHANGES_REQUESTED** [review]
@tari01: You remove a lot of code in this PR without properly documenting the underlying reasonings. Please split out those removals into extra commits, maybe even an extra PR and document more thoroughly the why. Thanks.

File: tests/print-to.h (hunk 2 above)
Comment (tari01) [reply tests/print-to.h:55]: Unused file - added separate commit.

File: tests/test-menu-appointments.cpp (hunk 3 above)
Comment (tari01) [reply tests/test-menu-appointments.cpp:183]: Unused variables - added separate commit.

File: src/menu.cpp (hunk 1 above)
Comment (tari01) [reply src/menu.cpp:113]: Comparing variables with same value - removed block in separate commit.

Comment (sunweaver) [reply src/menu.cpp:113]: @tari01 maybe it should read

```
            const bool a_after_today = (a.begin > end_of_day) || (a.end > end_of_day);
            const bool b_after_today = (b.begin > end_of_day) || (b.end > end_of_day);
            if (a_after_today != b_after_today)
                return a_after_today;
```

????

**Review by sunweaver — COMMENTED** [review]
Please revisit the a_after_today / b_after_today code block removal. I proposed an alternative solution / fix in the inline discusssion.

File: src/menu.cpp (hunk 1 above)
Comment (tari01) [reply src/menu.cpp:113]: > @tari01 maybe it should read
> 
> ```
>             const bool a_after_today = (a.begin > end_of_day) || (a.end > end_of_day);
>             const bool b_after_today = (b.begin > end_of_day) || (b.end > end_of_day);
>             if (a_after_today != b_after_today)
>                 return a_after_today;
> ```
> 
> ????

I've tried all sorts of permutations of a and b, but it breaks the appointment ordering. I have a feeling this is a leftover from a previous approach that was replaced by adding more code blocks to this section. As things are now, the appointments are properly ordered - both in unit tests and the live installation.

**Comment by sunweaver** [comment]
@tari01 Hi! Any new status on this?

**Comment by tari01** [comment]
> @tari01 Hi! Any new status on this?

Yes, I commented:

https://github.com/AyatanaIndicators/ayatana-indicator-datetime/pull/86#discussion_r772293867

**Review by sunweaver — APPROVED** [review]
Ok, then. Let's merge this.

**Pull request merged**